File: backend/app/core/upgrades/curses/shattered_bullet_2.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class ShatteredBullet2(UpgradeImplementation):
    id = 'shattered_bullet_2'
    name = 'Shattered Bullet 2'
    description = '-3 bullet damage.'
    rarity = 'rare'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -3
        appliers.modifiers.add_multiplier('bullet', 'damage', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/shattered_bullet_2.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const ShatteredBullet2Def: UpgradeDef = {
  id: "shattered_bullet_2",
  name: "Shattered Bullet 2",
  description: "-3 bullet damage.",
  rarity: RarityID.Rare,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Bullet,
  fieldInTargetClass: UpgradeStatID.Damage,
  value: -3,
  isMultiplier: false,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class ShatteredBullet2 extends Upgrade {}
